react/features/rtcstats/RTCStats.ts

```diff
@@ -103,29 +105,50 @@ class RTCStats {
      */
     init(options: InitOptions) {
 
-        const { endpoint, meetingFqn, useLegacy, pollInterval, sendSdp } = options;
-
-        const traceOptions = {
-            endpoint,
-            meetingFqn,
-            onCloseCallback: this.handleTraceWSClose.bind(this),
-            useLegacy
-        };
-
-        this.trace = traceInit(traceOptions);
-        if (!this.isPeerConnectionWrapped) {
-            const rtcstatsOptions = {
-                connectionFilter,
-                pollInterval,
-                useLegacy,
-                sendSdp,
-                eventCallback: this.handleRtcstatsEvent.bind(this)
+        const { useLibRtcStats, meetingFqn } = options;
+
+        if (useLibRtcStats) {
+            logger.info('setting up libRtcStats');
+
+            const rtcOptions = {
+                meetingName: meetingFqn,
+                eventCallback: this.handleRtcstatsEvent.bind(this),
+                onCloseCallback: this.handleTraceWSClose.bind(this)
             };
 
-            const statsEntryCallback = { statsEntry: this.statsEntry.bind(this) };
+            JitsiMeetJS.setRtcStatsMeetingFqn(rtcOptions);
+
+            JitsiMeetJS.getRtcStatsTrace().then((traceF: any) => {
```

Comment:
nit. await would work as well

Reply:
Point taken. But TBH I'm not convinced that in this case, even though it would be less characters,  it would make the code more readable.

Please let me know if you want me to change it to await.